xamarin/_Shared/LoginPage.xaml.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using AsyncTask = System.Threading.Tasks.Task;
+using Realms.Sync;
+using Xamarin.Forms;
+
+namespace RealmTemplateApp
+{
+    public partial class LoginPage : ContentPage
+    {
+        private string email;
+        private string password;
+        private bool isLoggingIn;
+
+        public LoginPage()
+        {
+            InitializeComponent();
+        }
+
+        protected override void OnAppearing()
+        {
+            base.OnAppearing();
+            txtEmail.Text = "";
+            txtPassword.Text = "";
+            isLoggingIn = false;
+        }
+
+        private async void Main_Button_Clicked(object sender, EventArgs e)
+        {
+            if (isLoggingIn)
+            {
+                await DoLogin();
+            }
+            else
+            {
+                await RegisterUser();
+            }
+        }
+
+        private async AsyncTask DoLogin()
+        {
+            User user;
```

Comment:
It's not needed to declare this variable here, as it's used only inside the try-catch. Actually if you're not using it, you can simply not save the value returned by LoginAsync